contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -19,6 +19,19 @@ import { OVM_DeployerWhitelist } from "../precompiles/OVM_DeployerWhitelist.sol"
 
 /**
  * @title OVM_ExecutionManager
+ * @dev The Execution Manager (EM) is the core of our OVM implementation, and provides a sandboxed 
+ * environment allowing us to execute OVM transactions deterministically on either Layer 1 or 
+ * Layer 2. 
+ * The EM's run() function is the first function called during the execution of any
+ * transaction on L2. 
+ * For each context-dependent EVM operation the EM has a function which implements a corresponding 
+ * OVM operation, which will read state from the State Manager contract.
```

Comment:
nit: some of these do not read from the SM, instead reading from EM storage (messageContext and globalContext).  Not sure the succinct way to express.